### PR TITLE
feat: add benchmarking mode

### LINKED ISSUE #235: [TASK] Update the manager to handle benchmark mode.

### PR DESCRIPTION
**Pull request by shekohex**
Closes #235

### CODE REVIEW

File: blueprints/incredible-squaring/src/main.rs [hunk 1]
```diff
@@ -25,22 +25,29 @@ use std::sync::Arc;
 
 #[async_trait::async_trait]
 trait GadgetRunner {
```
Comment (drewstone) [inline blueprints/incredible-squaring/src/main.rs:27]: @shekohex can we move these traits into the SDK?

File: blueprints/incredible-squaring/src/main.rs [hunk 2]
```diff
@@ -75,26 +82,38 @@ impl GadgetRunner for TangleGadgetRunner {
 
         Ok(())
     }
+
+    async fn benchmark(&self) -> Result<()> {
+        let env = gadget_sdk::env::load(None)?;
+        let client = env.client().await?;
+        let signer = env.first_signer()?;
+        let xsquare_summery = blueprint::xsquare_benchmark();
```
Comment (drewstone) [inline blueprints/incredible-squaring/src/main.rs:90]:
```suggestion
        let xsquare_summary = blueprint::xsquare_benchmark();
```

Comment (drewstone) [reply blueprints/incredible-squaring/src/main.rs:90]: Spelling error, please check your spelling often @shekohex

File: blueprints/incredible-squaring/src/main.rs [hunk 3]
```diff
@@ -155,13 +173,21 @@ impl GadgetRunner for EigenlayerGadgetRunner {
 
         Ok(())
     }
+
+    async fn benchmark(&self) -> Result<()> {
+        let env = gadget_sdk::env::load(Some(Protocol::Eigenlayer))?;
+        let client = env.client().await?;
+        let signer = env.first_signer()?;
+        let xsquare_summery = blueprint::xsquare_benchmark();
```
Comment (drewstone) [inline blueprints/incredible-squaring/src/main.rs:181]:
```suggestion
        let xsquare_summary = blueprint::xsquare_benchmark();
```

**Review by drewstone — CHANGES_REQUESTED** [review]

File: blueprints/incredible-squaring/src/main.rs (hunk 1 above)
Comment (shekohex) [reply blueprints/incredible-squaring/src/main.rs:27]: Looks like a good idea, will move it to the SDK.

**Review by drewstone — APPROVED** [review]

**Pull request merged**